com.unity.render-pipelines.high-definition/Runtime/Lighting/Light/HDGpuLightsBuilder.Jobs.cs

```diff
@@ -0,0 +1,910 @@
+                var shadowDimmerVal = lightRenderData.shadowDimmer;
+                var volumetricShadowDimmerVal = lightRenderData.volumetricShadowDimmer;
+                float shadowDistanceFade = HDUtils.ComputeLinearDistanceFade(distanceToCamera, Mathf.Min(globalConfig.maxShadowFadeDistance, lightsShadowFadeDistance));
+                lightData.shadowDimmer = shadowDistanceFade * shadowDimmerVal;
+                lightData.volumetricShadowDimmer = shadowDistanceFade * volumetricShadowDimmerVal;
+
+                // We want to have a colored penumbra if the flag is on and the color is not gray
+                var shadowTintVal = lightRenderData.shadowTint;
+                bool penumbraTintVal = lightRenderData.penumbraTint && ((shadowTintVal.r != shadowTintVal.g) || (shadowTintVal.g != shadowTintVal.b));
+                lightData.penumbraTint = penumbraTintVal ? 1.0f : 0.0f;
+                if (penumbraTintVal)
+                    lightData.shadowTint = new Vector3(Mathf.Pow(shadowTintVal.r, 2.2f), Mathf.Pow(shadowTintVal.g, 2.2f), Mathf.Pow(shadowTintVal.b, 2.2f));
+                else
+                    lightData.shadowTint = new Vector3(shadowTintVal.r, shadowTintVal.g, shadowTintVal.b);
+
+                //Value of max smoothness is derived from Radius. Formula results from eyeballing. Radius of 0 results in 1 and radius of 2.5 results in 0.
+                float maxSmoothness = Mathf.Clamp01(1.1725f / (1.01f + Mathf.Pow(1.0f * (shapeRadiusVal + 0.1f), 2f)) - 0.15f);
+                // Value of max smoothness is from artists point of view, need to convert from perceptual smoothness to roughness
+                lightData.minRoughness = (1.0f - maxSmoothness) * (1.0f - maxSmoothness);
+                lightData.shadowMaskSelector = Vector4.zero;
+
+                if (processedEntity.isBakedShadowMask)
+                {
+                    lightData.shadowMaskSelector[visibleLightBakingOutput.occlusionMaskChannel] = 1.0f;
+                    lightData.nonLightMappedOnly = visibleLightShadowCasterMode == LightShadowCasterMode.NonLightmappedOnly ? 1 : 0;
+                }
+                else
+                {
+                    // use -1 to say that we don't use shadow mask
+                    lightData.shadowMaskSelector.x = -1.0f;
+                    lightData.nonLightMappedOnly = 0;
+                }
+            }
+
+            private void ConvertLightToGPUFormat(
+                int outputIndex, int lightIndex,
+                LightCategory lightCategory, GPULightType gpuLightType, LightVolumeType lightVolumeType)
+            {
+                var light = visibleLights[lightIndex];
+                var processedEntity = processedEntities[lightIndex];
+                var lightData = new LightData();
+                ref HDLightRenderData lightRenderData = ref GetLightData(processedEntity.dataIndex);
+
+                ConvertLightToGPUFormat(
```

Comment:
I might be missing something (will need to chat via slack when back), but isn't most of what is computed below on lightData already computed by this function on the value passed as ref?

Reply:
Yes good catch. When I moved the contents into the function i forgot to delete this. Good catch!